src/artifacts/entities/artifact.entity.ts

```diff
@@ -44,8 +44,7 @@ export abstract class Artifact extends PrincipalScopedEntity {
   @ManyToOne()
   message?: Ref<Message>;
 
-  @Index()
-  @Property()
+  @Property({ unique: true })
```

Comment:
Shouldn't we implement a retry for the save? This value is autogenerated so this should not fail on the unique constraint.